pkg/server/api_server.go

```diff
@@ -115,6 +116,9 @@ func NewAPIServer(options ServerOptions) *ApiServer {
 	s.httpServer = &http.Server{
 		Addr:    options.ServerConfig.BindAddress,
 		Handler: mainHandler,
+		TLSConfig: &tls.Config{
+			MinVersion: options.ServerConfig.MinTLSVersion,
+		},
```

Comment:
Do we need to set this conditionally based on whether https is enabled?

Reply:
I don't think so.
See the description of the TLSConfig field in http#Server: https://pkg.go.dev/net/http#Server :

```
// TLSConfig optionally provides a TLS configuration for use
	// by ServeTLS and ListenAndServeTLS. Note that this value is
	// cloned by ServeTLS and ListenAndServeTLS, so it's not
	// possible to modify the configuration with methods like
	// tls.Config.SetSessionTicketKeys. To use
	// SetSessionTicketKeys, use Server.Serve with a TLS Listener
	// instead.
	TLSConfig *[tls](https://pkg.go.dev/crypto/tls).[Config](https://pkg.go.dev/crypto/tls#Config)
```

So my understanding is that is only used when ListenAndServeTLS or ServeTLS are called, and not when ListenAndServe or Serve are called.

Reply:
That make sense. Thanks for checking.